youtube_dl/extractor/kaltura.py

```diff
@@ -139,6 +140,13 @@ def _extract_url(webpage):
                 url = smuggle_url(url, {'service_url': service_url.group(1)})
             return url
 
+        # Check for an iframe, which may require redirection.
+        mobj = re.search(
+            r"<iframe[^>]+src=['\"](?P<url>(https?:)?//www\.kaltura\.com/[^'\"]+)['\"]",
```

Comment:
You should extract `partner_id` and `entry_id` and return `kaltura:...` shortcut.